IMDb.js

```diff
@@ -60,98 +64,78 @@ function getSearchResults(doc, checkOnly) {
 	return found ? items : false;
 }
 
-
 function doWeb(doc, url) {
 	if (detectWeb(doc, url) == "multiple") {
 		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
-				return true;
+				return;
 			}
 			var articles = [];
 			for (var i in items) {
 				articles.push(i);
 			}
 			ZU.processDocuments(articles, scrape);
 		});
-	} else {
+	}
+	else {
 		scrape(doc, url);
 	}
 }
 
-
 function scrape(doc, url) {
 	var translator = Zotero.loadTranslator('web');
 	// Embedded Metadata
 	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
-	//translator.setDocument(doc);
+	// translator.setDocument(doc);
 	
 	translator.setHandler('itemDone', function (obj, item) {
-		var titleWrapper = ZU.xpath(doc, '//div[contains(@class, "title_wrapper")]');
-		var title = ZU.xpathText(titleWrapper, './h1/text()[1]');
-		if (title) {
-			item.title = title;
-		}
-		item.date = ZU.xpathText(titleWrapper, './/meta[@itemprop="datePublished"]/@content');
-		item.runningTime = ZU.xpathText(titleWrapper, './/time[@itemprop="duration"]');
-		item.genre = ZU.xpathText(titleWrapper, './/span[@itemprop="genre"]');
-		var origTitle = ZU.xpathText(titleWrapper, './/div[contains(@class, "originalTitle")]/text()[1]');
-		if (origTitle) {
-			addExtra(item, "original-title: "+origTitle);
-		}
-		var pageId = ZU.xpathText(doc, '//meta[@property="pageId"]/@content');
-		if (pageId) {
-			addExtra(item, "IMDb ID: "+pageId);
-		}
-		
-		var summary = ZU.xpath(doc, '//div[contains(@class, "plot_summary_wrapper")]');
+		let json = JSON.parse(text(doc, 'script[type="application/ld+json"]'));
+		item.title = json.name;// note that json only has the original title
+		item.date = "datePublished" in json ? json.datePublished : "";
```

Comment:
Is this the right way to check if the field is in the json data?

Reply:
You should be able to just do `item.date = json.datePublished;`. If it's undefined then it'll just be undefined.